### PR TITLE
fix: sourcemaps are not working when rebuilding during webpack watch

### PR DESCRIPTION
**Pull request by alan-agius4**
## I'm submitting a...

```
[x] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [X] Tests for the changes have been added

## Description

What this change brings is;
- `relocation` and `rewiring` of sourcemaps happens during flattening part.
- `ESM5` and `ESM2015` has inlined sourcemaps which conforms with `APF V6`
- Downleveling of `ESM2015` to `ESM5` has been removed in favor of an `ngc` parallel build this **improve the build time by approx 10%** in some cases. **This is also a step towards incremental builds**. 

If you compare the `yarn test` of this CI and the another `yarn test` in another CI it takes approx **60 seconds less** on 17 very small samples.

Closes: #785, #803 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/lib/ng-v5/entry-point/write-bundles.transform.ts [hunk 1]
```diff
@@ -67,7 +68,7 @@ async function writeFlatBundleFiles(destinationFiles: DestinationFiles, opts: Fl
   log.info('Bundling to UMD');
   await flattenToUmd({
     ...opts,
-    entryFile: fesm5,
+    entryFile: esm5,
```
Comment (alan-agius4) [inline src/lib/ng-v5/entry-point/write-bundles.transform.ts]: this is needed to keep the sourcemaps intact

**Comment by dherges** [comment]
This changes sourcemaps from external `.map` files to inline source maps. Is that right?

**Comment by alan-agius4** [comment]
For esm and esm2015 yes, for the others they are still external

**Comment by alan-agius4** [comment]
bump

**Comment by alan-agius4** [comment]
fixed conflicts from upstream

**Comment by dherges** [comment]
I think the fesm/umd bundles are missing the last line `//# sourceMappingURL=<filePath>.js.map`

File: src/lib/flatten/rollup.ts [hunk 2]
```diff
@@ -49,13 +52,29 @@ export async function rollupBundleFile(opts: RollupOptions): Promise<void> {
...
-  await bundle.write({
-    name: `${opts.moduleName}`,
-    file: opts.dest,
+  const result = await bundle.generate({
+    name: opts.moduleName,
     format: opts.format,
     amd: opts.amd,
     banner: '',
     globals: moduleId => umdModuleIdStrategy(moduleId, opts.umdModuleIds || {}),
     sourcemap: true
```
Comment (dherges) [inline src/lib/flatten/rollup.ts]: For rollup, we'd need to change the output options:

```diff
diff --git a/src/lib/flatten/rollup.ts b/src/lib/flatten/rollup.ts
index 67ee77b..1dc459f 100644
--- a/src/lib/flatten/rollup.ts
+++ b/src/lib/flatten/rollup.ts@@ -52,13 +52,16 @@ export async function rollupBundleFile(opts: RollupOptions): Promise<void[]> {
   });

   // Output the bundle to disk
+  const sourcemapFile = `${opts.dest}.map`;
   const result = await bundle.generate({
+    file: opts.dest,
     name: opts.moduleName,
     format: opts.format,
     amd: opts.amd,
     banner: '',
     globals: moduleId => umdModuleIdStrategy(moduleId, opts.umdModuleIds || {}),-    sourcemap: true
+    sourcemap: 'inline',
+    sourcemapFile
   });

   // relocate sourcemaps
@@ -76,5 +79,5 @@ export async function rollupBundleFile(opts: RollupOptions): Promise<void[]> {     }
   });

-  return Promise.all([outputJson(`${opts.dest}.map`, result.map), outputFile(opts.dest, result.code)]);
+  return Promise.all([outputJson(sourcemapFile, result.map), outputFile(opts.dest, result.code)]); }
```


Still then, we have to append the final line ourselves as [rollup does it in their `build` executable command](https://github.com/rollup/rollup/blob/master/bin/src/run/build.ts#L59-L63)

Comment (dherges) [reply src/lib/flatten/rollup.ts]: See https://github.com/rollup/rollup/issues/121

Comment (alan-agius4) [reply src/lib/flatten/rollup.ts]: Yeah, realised that. Thanks for your help. I’ll do these within the hour

Comment (alan-agius4) [reply src/lib/flatten/rollup.ts]: I pushed an update

File: src/lib/flatten/rollup.ts [hunk 3]
```diff
@@ -83,6 +83,6 @@ export async function rollupBundleFile(opts: RollupOptions): Promise<void[]> {
 
   // rollup doesn't add a sourceMappingURL
   // https://github.com/rollup/rollup/issues/121
-  result.code = `${result.code}//# sourceMappingURL=${sourcemapFile}`;
+  result.code = `${result.code}\n//# sourceMappingURL=${sourcemapFile}`;
```
Comment (alan-agius4) [inline src/lib/flatten/rollup.ts]: no need for this :P

File: src/lib/flatten/rollup.ts (hunk 2 above)
Comment (alan-agius4) [reply src/lib/flatten/rollup.ts]: bdw, kudos for realising that `sourceMappingURL` was not being added

File: src/lib/flatten/rollup.ts [hunk 4]
```diff
@@ -83,6 +83,6 @@ export async function rollupBundleFile(opts: RollupOptions): Promise<void[]> {
 
   // rollup doesn't add a sourceMappingURL
   // https://github.com/rollup/rollup/issues/121
-  result.code = `${result.code}\n//# sourceMappingURL=${sourcemapFile}`;
+  result.code = `${result.code}\n//# sourceMappingURL=${result.map.toUrl()}`;
```
Comment (alan-agius4) [inline src/lib/flatten/rollup.ts]: 👍

**Comment by alan-agius4** [comment]
Thanks

**Pull request merged**